plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java

```diff
@@ -687,6 +687,19 @@ public void testNestedUuid()
         assertQuery("SELECT int_t FROM test_nested_uuid WHERE row_t.uuid_t = UUID '406caec7-68b9-4778-81b2-a12ece70c8b1'", "VALUES 2");
     }
 
+    @Test
+    public void testPredicateOnNestedField()
+    {
+        assertUpdate("CREATE TABLE test_predicate_on_nested_field (int_t int, row_t row(uuid_t uuid, int_t int))");
```

Comment:
nit: How about using simple type (e.g. int or varchar) instead of uuid for improving the readability?